### ISSUE TITLE
Cross validation early stopping

### ISSUE THREAD

**Issue by segatrade** [microsoft/LightGBM#5683]
Now cross validation early stopping happen based on `mean`. But seems it's more correct to use minimum from all folds in iteration, if we want to choose `num_iterations` based on `best_iteration` for train model on complete dataset after cv. 

https://sites.google.com/site/lauraeppx/xgboost/cross-validation also seems @Laurae2 tell here about it

Is any way to change `lgbm.cv` from mean to min mode? Or only my own cv with usual `lgbm.train` calls?
Also if make my own - Does `lgbm.cv` have performance benefits than call several time `lgbm.train` that I can use? It load data 1 time or several?

**Comment by 3zhang** [comment]
I don't think so. It's possible that you have a fold whose error is monotonically decreasing but still higher than other folds whereas other folds do have their minimums in early rounds . Then choosing the worst error will always set the best iter to the total number of iters.
